File: KCD/HMCreateSlotItemCommand.m
//
//  HMCreateSlotItemCommand.m
//  KCD
//
//  Created by Hori,Masaki on 2014/03/22.
//  Copyright (c) 2014年 Hori,Masaki. All rights reserved.
//

#import "HMCreateSlotItemCommand.h"

#import "HMMemberMaterialCommand.h"
#import "HMUpdateSlotItemCommand.h"
#import "HMStoreCreateSlotItemHistoryCommand.h"


@implementation HMCreateSlotItemCommand
+ (void)load
{
	static dispatch_once_t onceToken;
	dispatch_once(&onceToken, ^{
		[HMJSONCommand registerClass:self];
	});
}

+ (BOOL)canExcuteAPI:(NSString *)api
{
	if([api isEqualToString:@"/kcsapi/api_req_kousyou/createitem"]) return YES;
	
	return NO;
}
- (id)init
{
	self = [super init];
	self = [[super class] compositCommandWithCommands:
			[HMMemberMaterialCommand new],
			[HMUpdateSlotItemCommand new],
			[HMStoreCreateSlotItemHistoryCommand new],
			nil];
	return self;
}

@end

File: KCD/HMCreateSlotItemCommand.swift
//
//  HMCreateSlotItemCommand.swift
//  KCD
//
//  Created by Hori,Masaki on 2015/01/15.
//  Copyright (c) 2015年 Hori,Masaki. All rights reserved.
//

import Cocoa

class HMCreateSlotItemCommand: HMCompositCommand, NSCopying {
	required override init(commandArray: [AnyObject]) {
		super.init(commandArray: commandArray)
	}
	
	override class func load() {
		// Mavericksにおいてload()のタイミングでNSArrayを生成することが出来ないため、遅延させる
		dispatch_after(dispatch_time(DISPATCH_TIME_NOW, Int64(1 * Double(NSEC_PER_MSEC))), dispatch_get_main_queue()) {
			HMJSONCommand.registerInstance(self(commandArray: self.commands()))
		}
	}
	override class func canExcuteAPI(api: String) -> Bool {
		return api == "/kcsapi/api_req_kousyou/createitem"
	}
	
	override func copyWithZone(zone: NSZone) -> AnyObject {
		return self.dynamicType.init(commandArray: self.dynamicType.commands())
	}
	
	class func commands() -> [AnyObject] {
		return [HMMemberMaterialCommand(),
			HMUpdateSlotItemCommand(),
			HMStoreCreateSlotItemHistoryCommand()]
	}
	
	override func execute() {
		NSThread.sleepForTimeInterval(6.5)
		super.execute()
	}
}
